plugin/evm/warp_backend.go

```diff
@@ -0,0 +1,82 @@
+// (c) 2019-2020, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package evm
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/cache"
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/database/prefixdb"
+	"github.com/ava-labs/avalanchego/database/versiondb"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow"
+	"github.com/ava-labs/avalanchego/utils/hashing"
+	"github.com/ava-labs/avalanchego/vms/platformvm/teleporter"
+)
+
+var (
+	_ WarpBackend = &warpBackend{}
+
+	dbPrefix = []byte("warp")
+)
+
+// WarpBackend keeps track of messages that are accepted by the warp precompiles and add them into a database.
+// The backend is also used to query for warp message signatures by the signature request handler.
+type WarpBackend interface {
+	// AddMessage is called in the precompile OnAccept, to add warp messages into the database.
+	AddMessage(ctx context.Context, unsignedMessage *teleporter.UnsignedMessage) error
+
+	// GetSignature returns the signature of the requested message hash.
+	GetSignature(ctx context.Context, messageHash ids.ID) ([]byte, error)
```

Comment:
nit:
```suggestion
	// GetSignature returns the signature of the requested messageID.
	GetSignature(ctx context.Context, messageID ids.ID) ([]byte, error)
```

we typically use `ID` in avalanchego and subnet-evm instead of hash, so we should make this change throughout the file